### PR TITLE
templates: ignore `metafile.*` files

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
I originally just wanted to ignore the `metafile.*` files, but since the entire build directory is generated I figured it would be better to ignore it entirely 🤷‍♂️

---

Supersedes #7090

### CODE REVIEW

File: templates/arc/.gitignore [hunk 1]
```diff
@@ -1,10 +1,8 @@
-.DS_Store
 node_modules
 
 /.cache
-/server/index.mjs
-/server/index.mjs.map
 /public/build
+/server
```
Comment (lpsinger) [inline templates/arc/.gitignore:5]: The file config.arc is under version control in the /server directory. If you want to ignore the whole server directory, then I suggest that you move the `@aws` section to app.arc and remove config.arc.

**Review by lpsinger — CHANGES_REQUESTED** [review]

File: templates/cloudflare-pages/.gitignore [hunk 2]
```diff
@@ -1,8 +1,6 @@
-.DS_Store
 node_modules
 
 /.cache
-/functions/\[\[path\]\].js
-/functions/\[\[path\]\].js.map
+/functions
```
[the remark this answers is not in the document]
Comment (MichaelDeBoey) [reply templates/cloudflare-pages/.gitignore:4]: By default there are no other functions except the generated ones

People can also always unignore certain files if they want to

**Review by lpsinger — APPROVED** [review]

File: templates/cloudflare-pages/.gitignore (hunk 2 above)
[the remark this answers is not in the document]
Comment (pcattori) [reply templates/cloudflare-pages/.gitignore:4]: I agree that we should try to target only the Remix-generated functions in the gitignore

[the remark this answers is not in the document]
Comment (MichaelDeBoey) [reply templates/cloudflare-pages/.gitignore:4]: @pcattori I assume only if we by default choose for another folder than `build` (which is the default build folder)?

**Review by pmbanugo — APPROVED** [review]
✨

**Review by pcattori — APPROVED** [review]

**Pull request merged**